marquez_airflow/dag.py

```diff
@@ -113,53 +115,91 @@ def create_dagrun(self, *args, **kwargs):
+
                 if extractor:
                     self.log.debug(
                         f"Using extractor '{extractor}' to apply partial "
-                        f"metadata updates for operator '{ti.operator}' "
+                        f"metadata updates for operator '{task.__class__}' "
                         f"in DAG '{self.dag_id}'."
                     )
 
                     steps_meta = add_airflow_info_to(
-                        ti.operator,
-                        extractor(ti.operator).extract_on_complete(ti)
-                    )
+                        task,
+                        extractor(task).extract_on_complete(ti))
                     self.log.debug(f'steps_meta: {steps_meta}')
 
-                    # FIXME: replace with update api
                     for step in steps_meta:
-                        self.get_marquez_client().create_job(
+                        self.log.info(f'step: {step}')
+
+                        marquez_run_id = self._get_marquez_run_id(
+                            ti, dagrun, kwargs)
+                        self.log.info(f'marquez_run_id: {marquez_run_id}')
+
+                        location = (step.location or self._get_location(ti))
+
+                        self.register_datasets(step.inputs, marquez_run_id)
+                        inputs = list(
+                            map(lambda input_data: {
+                                'namespace': self.marquez_namespace,
+                                'name': input_data.name
+                            }, step.inputs))
+                        self.log.info(f'inputs: {inputs}')
+
+                        self.register_datasets(step.outputs, marquez_run_id)
+                        outputs = list(
+                            map(lambda output_data: {
+                                'namespace': self.marquez_namespace,
+                                'name': output_data.name
+                            }, step.outputs))
+                        self.log.info(f'outputs: {outputs}')
+
+                        self.get_or_create_marquez_client().create_job(
+                            namespace_name=self.marquez_namespace,
                             job_name=step.name,
                             job_type=JobType.BATCH,
-                            location=(step.location or self._get_location(ti)),
-                            input_dataset=step.inputs,
-                            output_dataset=step.outputs,
+                            location=location,
+                            input_dataset=inputs,
+                            output_dataset=outputs,
+                            description=step.description,
                             context=step.context,
-                            description=self.description,
-                            namespace_name=self.marquez_namespace)
+                            run_id=marquez_run_id)
+
+                        self.log.info(
```

Comment:
I'd would be explicit and add the job name and run ID in the log message.

Reply:
done.